java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/SinkStreamObserver.java

```diff
@@ -188,25 +197,29 @@ private void cleanup() {
         if (sink != null) {
             sink.drop();
         }
+        ConnectorNodeMetrics.decActiveSinkConnections(connectorType, "node1");
     }
 
-    private void bindSink(SinkConfig sinkConfig, ConnectorServiceProto.SinkPayloadFormat format) {
-        tableSchema = TableSchema.fromProto(sinkConfig.getTableSchema());
+    private void bindSink(
+            long sinkId, SinkConfig sinkConfig, ConnectorServiceProto.SinkPayloadFormat format) {
```

Comment:
nit: we don't have to pass the sink id here only for setting `this.sink_id = sink_id`. We can avoid passing the sink id here and set `this.sink_id` after or before calling this method.